volumes:
  local_postgres_data: {}
  local_pgadmin: {}
  local_redis_data: {}

services:
  django:
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: taskminder_local_django
    container_name: taskminder_local_django
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - .:/app:z
    env_file:
      - ./.envs/.local/.django
      - ./.envs/.local/.postgres
    ports:
      - '8000:8000'
    command: /start

  postgres:
    image: postgres:16
    container_name: taskminder_local_postgres
    volumes:
      - local_postgres_data:/var/lib/postgresql/data
    env_file:
      - ./.envs/.local/.postgres
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}" ]
      interval: 3s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4:8.12
    container_name: taskminder_local_pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@pgadmin.com
      PGADMIN_DEFAULT_PASSWORD: password
      PGADMIN_LISTEN_PORT: 80
    ports:
      - 15432:80
    volumes:
      - local_pgadmin:/var/lib/pgadmin
    depends_on:
      - postgres

  mailpit:
    image: docker.io/axllent/mailpit:latest
    container_name: taskminder_local_mailpit
    ports:
      - "8025:8025"
